polyply/src/dispatch_gmx.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+import vermouth
+import os
+
+class GMXError(Exception):
+    """
+    Class for handeling GROMACS errors. This class only
+    prints the essentail part to the error message when
+    raised.
+    """
+
+    def __init__(self, stderr):
+        lines = stderr.decode("utf-8")
+        message = "GROMACS terminated with the following error:\n"
+        message = message + lines
+        super().__init__(message)
+
+
+class DispatchGMX:
+    """
+    Abstract base class for running gromacs from
```

Comment:
Nothing abstract about this class.

Also, missing docs on what the attributes/arguments are. Is gmx_path e.g. `/usr/local/gromacs-2020//bin/gmx` and gmx_exe `grompp`?
Does this work with GMX < 5?